dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/SubmissionFormConverter.java

```diff
@@ -0,0 +1,147 @@
+import org.dspace.app.rest.model.SubmissionFormRest;
+import org.dspace.app.rest.model.SubmissionVisibilityRest;
+import org.dspace.app.rest.model.VisibilityEnum;
+import org.dspace.app.rest.utils.AuthorityUtils;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.app.util.DCInput;
+import org.dspace.app.util.DCInputSet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the converter from/to the DCInputSet in the DSpace API data model and
+ * the REST data model
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+@Component
+public class SubmissionFormConverter extends DSpaceConverter<DCInputSet, SubmissionFormRest> {
+
+	private static final String INPUT_TYPE_ONEBOX = "onebox";
+	private static final String INPUT_TYPE_NAME = "name";
+	private static final String INPUT_TYPE_LOOKUP = "lookup";
+	private static final String INPUT_TYPE_LOOKUP_NAME = "lookup-name";
+
+	@Autowired
+	private Utils utils;
+
+	@Autowired
+	private AuthorityUtils authorityUtils;
+
+	@Override
+	public SubmissionFormRest fromModel(DCInputSet obj) {
+		SubmissionFormRest sd = new SubmissionFormRest();
+		sd.setName(obj.getFormName());
+		DCInput[] step = obj.getFields();
+		List<SubmissionFormFieldRest> fields = getPage(step);
+		sd.setFields(fields);
+		return sd;
+	}
+
+	private List<SubmissionFormFieldRest> getPage(DCInput[] page) {
+		List<SubmissionFormFieldRest> fields = new LinkedList<SubmissionFormFieldRest>();
+		for (DCInput dcinput : page) {
+			fields.add(getField(dcinput));
+		}
+		return fields;
+	}
+
+	private SubmissionFormFieldRest getField(DCInput dcinput) {
+		SubmissionFormFieldRest inputField = new SubmissionFormFieldRest();
+		List<SelectableMetadata> selectableMetadata = new ArrayList<SelectableMetadata>();
+
+		inputField.setLabel(dcinput.getLabel());
+		inputField.setHints(dcinput.getHints());
+		inputField.setMandatoryMessage(dcinput.getWarning());
+		inputField.setMandatory(dcinput.isRequired());
+		inputField.setScope(ScopeEnum.fromString(dcinput.getScope()));
+		inputField.setVisibility(new SubmissionVisibilityRest(
+				VisibilityEnum.fromString(dcinput.isReadOnly("submission") ? "read-only" : null),
+				VisibilityEnum.fromString(dcinput.isReadOnly("workflow") ? "read-only" : null)));
+		inputField.setRepeatable(dcinput.isRepeatable());
+
+		SubmissionFormInputTypeRest inputRest = new SubmissionFormInputTypeRest();
+
+		inputRest.setRegex(dcinput.getRegex());
+
+		if (!StringUtils.equalsIgnoreCase(dcinput.getInputType(), "qualdrop_value")) {
+			// value-pair and vocabulary are a special kind of authorities
+			String inputType = dcinput.getInputType();
+
+			SelectableMetadata selMd = new SelectableMetadata();
+			if (authorityUtils.isChoice(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier())) {
+				inputRest.setType(
+						getPresentation(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier(), inputType));
```

Comment:
Just a question: Shouldn't authority presentation information be retrieved from the corresponding authority endpoint?

Reply:
It is a concept related to metadata type, that in certain case have to modify the presentation. The authority endpoint is not presentation aware because expone only information related the structure and the data.